Tests/MongoSwiftTests/CodecTests.swift

```diff
@@ -509,4 +486,208 @@ final class CodecTests: XCTestCase {
         let values2 = [BasicStruct(int: 1, string: "hello"), nil]
         expect(try encoder.encode(values2)).to(equal([["int": 1, "string": "hello"], nil]))
     }
+
+    struct AnyBsonStruct: Codable {
+        let x: AnyBsonValue
+
+        init(_ x: BsonValue) {
+            self.x = AnyBsonValue(x)
+        }
+    }
+
+    // test encoding/decoding AnyBsonValues with BsonEncoder and Decoder
```

Comment:
I realize it's not an AnyBsonValue, but is there a test case for `nil` (and/or do we need one)?